lib/registry.ex

```diff
@@ -72,9 +72,10 @@ defmodule Registry do
       # Prints #PID<...> where the pid is for the process that called register/3 above
       #=> :ok
 
-  Keep in mind dispatching happens in the process that calls `dispatch/3`, so
-  the callback is executed there.  The registered processes are not involved in
-  dispatching unless such is done explicitly. In the example, if there is a
+  Keep in mind that if the registry is not partitioned then dispatching happens
```

Comment:
What about:

```
Dispatching happens in the process that calls `dispatch/3` either serially
or concurrently in case of multiple partitions. The registered processes are
not involved in dispatching unless such is done explicitly.

Furthermore, if there is a failure when dispatching, due to a bad registration,
dispatching will always fail and the registered process is not notified. Therefore
let's make sure we at last wrap and report those errors:
```